app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author, class_name: 'User'
+  has_many :comment
+  has_many :like
+
+  def update_counter(user_id = author_id)
+    user = User.find(user_id)
+    user.posts_counter = Post.where(author_id: user.id).count
+    user.save
+  end
```

Comment:
- [ ] Your `update_counter` method for the Post model gets the user's posts but is not updating(incrementing) the posts counter. Please add the necessary logic to fix this.
For instance, you can try using `user.increment!(:posts_counter)` or `user.posts_counter += 1`

[OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private

```
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    # logic here
  end

```

Reply:
Hi @petudeveloper. Thank you for your suggestions! But my method already returns the number of posts a user has. It uses the `count` method, to count every post where `author_id: user.id`. Why should I change that?